apps/prometheus/deployers/prometheus.yml

```diff
@@ -1,67 +1,160 @@
+        action: keep
+      - source_labels: [__meta_kubernetes_pod_name]
+        action: replace
+        target_label: kubernetes_pod_name
+---
+# prometheus-deployment
+apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: prometheus
-  labels:
-    lkey: lvalue
-    openebs.io/target-affinity: prometheus
+  name: openebs-prometheus
 spec:
   replicas: 1
-  selector: 
-    matchLabels:
-      lkey: lvalue
-  template: 
+  template:
     metadata:
-      labels: 
+      labels:
         lkey: lvalue
-        openebs.io/target-affinity: prometheus
     spec:
+      serviceAccountName: app-prometheus-operator
       containers:
-        - resources:
-            limits:
-              cpu: 0.5
-          name: prometheus
-          image: prom/prometheus
-          imagePullPolicy: IfNotPresent 
-          #args:
-          env:
-            - name: MYSQL_ROOT_PASSWORD
-              value: k8sDem0
+        - name: prometheus
+          image: prom/prometheus:v2.3.0
+          args:
+            - "--config.file=/etc/prometheus/conf/prometheus.yml"
+            # Metrics are stored in an emptyDir volume which
+            # exists as long as the Pod is running on that Node.
+            # The data in an emptyDir volume is safe across container crashes.
+            - "--storage.tsdb.path=/prometheus"
+            # How long to retain samples in the local storage.
+            - "--storage.tsdb.retention=$(STORAGE_RETENTION)"
           ports:
-            - containerPort: 9000
-              name: prometheus
+            - containerPort: 9090
+          env:
+            # environment vars are stored in prometheus-env configmap. 
+            - name: STORAGE_RETENTION
+              valueFrom:
+                configMapKeyRef:
+                  name: openebs-prometheus-tunables
+                  key: storage-retention
+          resources:
+            requests:
+              # A memory request of 250M means it will try to ensure minimum
+              # 250MB RAM .
+              memory: "128M"
+              # A cpu request of 128m means it will try to ensure minimum
+              # .125 CPU; where 1 CPU means :
+              # 1 *Hyperthread* on a bare-metal Intel processor with Hyperthreading
+              cpu: "128m"
+            limits:
+              memory: "700M"
+              cpu: "500m"
+          
           volumeMounts:
-            - mountPath: /etc/prometheus/conf
-              name: data-vol
+            # prometheus config file stored in the given mountpath
+            - name: prometheus-server-volume
+              mountPath: /etc/prometheus/conf
+            # metrics collected by prometheus will be stored at the given mountpath.
+            - name: prometheus-storage-volume
+              mountPath: /prometheus
       volumes:
-        - name: data-vol
-          persistentVolumeClaim:
-            claimName: testclaim
+        # Prometheus Config file will be stored in this volume 
```

Comment:
@harshita-sharma-mayadata   OpenEBS PVC is not used here. Need to include testclaim

Reply:
Prometheus data should be directed to cstor/jiva volume